### ISSUE TITLE
During make test, redirect stderr to a file

### ISSUE THREAD

**Issue by qualitesys** [openssl/openssl#10736]
I wish to collect the errors produced by the make test process.
Is it possible to configure the redirection of stderr to a file during "make test" ?

Thanks for your help

**Comment by mspncp** [comment]
I assume you are testing on linux or some other unix-ish operating system?

In the shell, you can redirect stdout and and stderr using the `>` operator. If a specific test is failing, say `test_xxx`, then it is better to rerun only this single test and 

    make TESTS=test_xxx V=1 test  > test.log  2>& 1

or simpler using `>&` to redirect both streams simultaneously (works in bash but not in bourne shell):

    make TESTS=test_xxx V=1 test  >& test.log 

For more information, see [this section of the INSTALL file](https://github.com/mspncp/openssl/blob/revamp-the-docs/INSTALL.md#test-openssl) (WIP, pull request #10545).

**Comment by dcr-eto** [comment]
Thanks for your quick answer, I'll try this way.

In my process I have added instrumentation routines inside the normal functions, that produce additional stderr outputs. It seams that the Ok/Ko test run status is confused by these additional outputs.

Is the stderr feed used to determine the status of a test ? If true I'll have to change my mind.
Thanks

**Comment by richsalz** [comment]
If you want to add your own output to the tests, use the ```TEST_NOTE``` macro.

**Comment by mspncp** [comment]
> Thanks for your quick answer, I'll try this way.
> In my process I have added instrumentation routines inside the normal functions, that produce additional stderr outputs. It seams that the Ok/Ko test run status is confused by these additional outputs.

_(Did you delete your post? I received it only via email)_

>Is the stderr feed used to determine the status of a test ? If true I'll have to change my mind.
Thanks

I'm not sure.  @levitte could tell you what triggers the error detection, but I think he is still on vacation. Maybe @richsalz knows the answer too.

**Comment by richsalz** [comment]
>delete your post?

Same here.

The return value from the main routine triggers the error/ok status.  Make sure you're not returning early.  To add logging, use the ```TEST_NOTE``` macro.  But the *output of the test is not parsed*

**Comment by mspncp** [comment]
> To add logging, use the `TEST_NOTE` macro.

Is this a Perl TAP feature? I did not find a single mention in the source tree. Also, a quick internet search did not yield anything useful.

**Comment by richsalz** [comment]
Part of our stuff, in ```testutil.h``` like ```TEST_int_eq``` et al.

**Comment by mspncp** [comment]
Oh, I see: you mean `TEST_note()`:

https://github.com/openssl/openssl/blob/e7b834b6bb0a32be694ebc8e614247c9af735c0f/test/testutil.h#L475

**Comment by mspncp** [comment]
```
msp@msppc:~$ man TEST_note
No manual entry for TEST_note
```

**Comment by richsalz** [comment]
oops, yeah TYPO  (I mean typo :)

**Comment by qualitesys** [comment]
Yes I use 2 github user names (1 personal and 1 pro). Openssl is a personal subject in fact.
For the moment I run a test session to verify the stderr interpretation in the Ok/Ko status check.
If it is confusing, I'll change my instrumentation code to generate output files other than stderr.
Thanks to mspncp and richsalz for your support !

**Comment by nhorman** [comment]
Marking as inactive, will be closed at the end of 3.4 dev barring further input
